components/nav/nav.tsx

```diff
@@ -0,0 +1,117 @@
+
+export function Nav() {
+  const rssLink = `${SiteConfig.site.siteUrl}/rss.xml`;
+  const { theme, setTheme } = useTheme();
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => setMounted(true), []);
+
+  return (
+    <header className="relative z-10 border-b border-accent-8">
+      <Container width="bleed">
+        <nav className="navWrapper flex justify-between items-center py-3 xl:p-0">
+          <Link href="/" className="flex items-center no-underline">
+            <Logo />
+            <span className="text-base text-foreground">
+              {SiteConfig.author.name}
+            </span>
+          </Link>
+
+          {/* TODO: make the links configurable */}
+          <div className="relative">
+            {mounted && (
+              <div className="flex gap-4 lg:hidden">
+                <button
+                  onClick={() => setTheme(theme === 'dark' ? 'light' : 'dark')}
+                  className="transition-colors text-accent-4 hover:text-accent-1"
+                >
+                  {theme === 'dark' ? <Moon /> : <Sun />}
+                </button>
+                <Hamburger />
+              </div>
+            )}
+
+            <nav className="hidden list-none md:flex md:items-center">
+              <NavItem dataTestId="workPageLink" title="Works" href="/works" />
+              <NavItem
+                dataTestId="articlesPageLink"
+                title="Articles"
+                href="/articles"
+              />
+              <NavItem dataTestId="notesPageLink" title="Notes" href="/notes" />
+              <NavItem dataTestId="aboutPageLink" title="About" href="/about" />
+
+              <NavItem
+                dataTestId="rssPageLink"
+                title={<Rss className="text-foreground" />}
+                href={rssLink}
+                external
+              />
+              <div>
+                {mounted ? (
+                  <button
+                    className="p-4 border-l border-accent-8 text-accent-4 hover:text-accent-1"
+                    onClick={() =>
+                      setTheme(theme === 'dark' ? 'light' : 'dark')
+                    }
+                  >
+                    {theme === 'dark' ? <Moon /> : <Sun />}
+                  </button>
+                ) : (
+                  <div className="p-4 border-l border-accent-8 text-accent-6">
+                    <Loader className="animate-spin" />
+                  </div>
+                )}
+              </div>
+            </nav>
+          </div>
+        </nav>
+      </Container>
+    </header>
+  );
+}
+
+function NavItem({
+  title,
+  href,
+  dataTestId,
+  external = false,
+}: {
+  title: string | ReactNode;
+  href: string;
+  dataTestId?: string;
+  external?: boolean;
+}) {
+  const classes =
+    'border-l border-accent-8 text-accent-4 hover:text-accent-1 p-4 uppercase';
+
+  if (external) {
+    return (
+      <a
+        data-test-id={dataTestId}
```

Comment:
Add `data-test-id` for Playwright testing

Reply:
Removed all data-test-ids 

[24018f3](https://github.com/upleveled/next-starter-peacock/pull/156/commits/24018f374b13b8104d6c67a18053d674b6218537)

Reply:
```js
    await page
      .locator('div:has-text("WorksArticlesNotesAbout") a')
      .nth(4)
      .getAttribute('href'),
```

- doesn't it make more sense to use `nav` here instead of `div`? it seems there are 2 `nav`s, be sure to write a good selector that will select the correct one
- I guess using the `.nth()` is also a bit brittle
- probably you can do a better version using just `page.locator(...).getAttribute('href')`